lib/SPIRV/SPIRVToOCL12.cpp

```diff
@@ -263,6 +278,9 @@ Instruction *SPIRVToOCL12Base::mutateAtomicName(CallInst *CI, Op OC) {
 std::string SPIRVToOCL12Base::mapAtomicName(Op OC, Type *Ty) {
   std::string Prefix = Ty->isIntegerTy(64) ? kOCLBuiltinName::AtomPrefix
                                            : kOCLBuiltinName::AtomicPrefix;
+  // Map fp atomic instructions to regular OpenCL built-ins.
+  if (isFPAtomicOpCode(OC))
+    return FPAtomicToOCLBuiltinName(OC);
   return Prefix += OCL12SPIRVBuiltinMap::rmap(OC);
```

Comment:
BTW, why can't we just extend `OCL12SPIRVBuiltinMap`?

Reply:
The problem with extending the `SPIRVMap<>` is caused by the fact that only one entry is allowed for each function name - mapping `add` to `AtomicFAddEXT` results in the overwrite of the `AtomicIAdd` entry.

Maybe we should define and use `atomic_f[add/min/max]` naming instead, similarly to what is done for `umax`/`umin`? @Fznamznon, @AlexeySotkin, what do you think here?

Reply:
As discussed offline, we will retain `atomic_[add/min/max]` naming for now, working out a unified strategy for float/unsigned atomic functions later.